neo/app/actor/NeoActor.scala

```diff
@@ -1,57 +1,107 @@
+class NeoActor(out: ActorRef, ws: WSClient, host: String, config: Config)
+              (implicit ec: ExecutionContext, implicit val materializer: Materializer) extends Actor with ActorLogging {
 
   import NeoActor._
-
   implicit val timeout: Timeout = Timeout(20.minutes)
 
   override def receive: Receive = {
     case json: JsValue =>
       json.validate[UserRequest].map { userRequest =>
-        val allRequest = generateCBerryRequest(userRequest)
+
+        // Generate json in berry format
         import RequestType._
+        val berryRequest = generateCBerryRequest(userRequest)
+        val url = "http://" + host + "/berry"
 
-        // Example of sending the non-slicing query using ask pattern
-        val sampleJson = allRequest(Sample)
-        val fAnswer = (berryClient ? sampleJson).mapTo[JsValue].map(json => Json.obj("key" -> "sample", "value" -> json))
-        fAnswer pipeTo out
+        // Sending requests on tweets samples
+        val sampleResponse : Future[StreamedResponse] =
+          ws.url(url).withMethod("POST").withBody(berryRequest(Sample)).stream()
+        
+        // Dealing with response on tweets samples: non-slicing
+        sampleResponse.map{ response =>
+          if(response.headers.status == 200){
+            try {
+              // Define sink of data stream
+              val sink = Sink.foreach[ByteString] { bytes =>
+                val json = Json.parse(bytes.utf8String)
+                out ! Json.obj("key" -> "sample", "value" -> json)
+              }
+              // streaming data pipeline
+              response.body
+                .via(Framing.delimiter(ByteString("\n"), maximumFrameLength = config.maxFrameLength, allowTruncation = true))
+                .runWith(sink)
+                .onFailure { case e => Logger.logger.error(e.toString) }
+            }
+            catch {
+              case e: Throwable => Logger.logger.error("stream receiving error: " + e.getMessage)
+            }
+          }
+          // Gateway error
+          else {
+            Logger.logger.error("Bad Gate Way")
+          }
+        }
 
-        // Example of sending the group of slicing ones using actor stream pattern
-        val groupTimePlaceTags = Seq(ByTime, ByPlace, ByHashTag).map(allRequest(_))
-        val json = JsObject(Seq(
+        // Sending group of slicing ones
+        val groupTimePlaceTags = Seq(ByTime, ByPlace, ByHashTag).map(berryRequest(_))
+        val batchJson = JsObject(Seq(
           "batch" -> JsArray(groupTimePlaceTags),
           "option" -> JsObject(Seq("sliceMillis" -> JsNumber(2000))
-          )))
-        berryClient.tell(BerryClient.Request(json, NeoTransformer("batch")), out)
+        )))
+
+        // Dealing with response on timing, geo and hashtags: slicing
+        val batchResponse: Future[StreamedResponse] =
+          ws.url(url).withMethod("POST").withBody(batchJson).stream()
+        batchResponse.map{ response =>
```

Comment:
it looks like the two response processing codes are very similar, can you make one function call that can be used in both cases by passing different parameters?